daemon/cluster/executor/container/controller.go

```diff
@@ -459,7 +460,7 @@ func (r *controller) waitReady(pctx context.Context) error {
 
 	ctnr, err := r.adapter.inspect(ctx)
 	if err != nil {
-		if !isUnknownContainer(err) {
+		if !errdefs.IsNotFound(err) {
```

Comment:
`isUnknownContainer` was only matching the error returned by `*ViewDB.GetByPrefix()`:

https://github.com/moby/moby/blob/f174059a2570a55b71261e869eda6f4a039a3141/container/view.go#L140

Is it safe?

Now, `errdefs.IsNotFound` will also matched the error returned here:

https://github.com/moby/moby/blob/c3217300e21ee3301faae0cfb0b7b4d576263292/daemon/container.go#L66

Reply:
Had a quick check; error didn't originally exist in this code, and was added in https://github.com/moby/moby/commit/00157a42d367eca1dc140a5638d41444ab7434ce.

But it looks like `containerNotFound` returns an `objNotFoundError , which would be `container` in this case as well; https://github.com/moby/moby/blob/9402ea1c8e9c7dacda47d5468d3a2b21dfc3432c/daemon/errors.go#L28-L41

So no change in behavior there, and from the earlier commit above (00157a42d367eca1dc140a5638d41444ab7434ce), the intent was to return a "not found" error, because previously it would not return an error (and no container).


So.. looks like we should be good 😄